src/components/dialog/dialog.tsx

```diff
@@ -0,0 +1,793 @@
+
+    this.#dragRAF ||= new SyncWithRAF();
+    this.#initialDragEvent = event;
+
+    // Add listeners
+    document.addEventListener("mousemove", this.#trackElementDragRAF, {
+      capture: true
+    });
+
+    document.addEventListener("mouseup", this.#handleDragEnd, {
+      capture: true,
+      passive: true
+    });
+  };
+
+  #trackElementDragRAF = (event: MouseEvent) => {
+    this.#dragRAF.perform(this.#trackElementDrag, () => {
+      // Improve drag UX by not selecting any button or clicking interactive elements
+      event.preventDefault();
+
+      // We remove the pointer-events and user-select properties after the first "mousemove", otherwise double clicking to select text would not work
+      this.#addDraggingClass();
+
+      this.#lastDragEvent = event;
+    });
+  };
+
+  #trackElementDrag = () => {
+    // - - - - - - - - - - - - - DOM read operations - - - - - - - - - - - - -
+    const currentDraggedDistanceX =
+      this.#lastDragEvent.clientX - this.#initialDragEvent.clientX;
+    const currentDraggedDistanceY =
+      this.#lastDragEvent.clientY - this.#initialDragEvent.clientY;
+
+    // - - - - - - - - - - - - - DOM write operations - - - - - - - - - - - - -
+    // Update total dragged distance in X
+    if (currentDraggedDistanceX !== 0) {
+      this.#draggedDistanceX += this.#isRTLDirection // Set dragged distance depending on RTL value
+        ? -currentDraggedDistanceX
+        : currentDraggedDistanceX;
+      setProperty(this.el, DIALOG_DRAGGED_X, this.#draggedDistanceX);
+    }
+
+    // Update total dragged distance in Y
+    if (currentDraggedDistanceY !== 0) {
+      this.#draggedDistanceY += currentDraggedDistanceY;
+      setProperty(this.el, DIALOG_DRAGGED_Y, this.#draggedDistanceY);
+    }
+
+    // Update last point
+    this.#initialDragEvent = this.#lastDragEvent;
+  };
+
+  #handleDragEnd = (event?: MouseEvent) => {
+    if (event) {
+      event.stopPropagation();
+    }
+    // Cancel RAF to prevent access to undefined references
+    if (this.#dragRAF) {
+      this.#dragRAF.cancel();
+    }
+
+    // Remove listeners
+    document.removeEventListener("mousemove", this.#trackElementDragRAF, {
+      capture: true
+    });
+
+    document.removeEventListener("mouseup", this.#handleDragEnd, {
+      capture: true
+    });
+
+    this.#removeDraggingClass();
+
+    // Free the memory
+    this.#dragRAF = null;
+    this.#initialDragEvent = null;
+    this.#lastDragEvent = null;
+  };
+
+  #showMethod = () => {
+    if (this.modal) {
+      this.#dialogRef.showModal();
+      document.addEventListener("click", this.#evaluateClickOnDocument, {
+        capture: true
+      });
+    } else {
+      this.#dialogRef.show();
+    }
+  };
+
+  #closeHandler = () => {
+    this.hidden = true;
+  };
+
+  #evaluateClickOnDocument = (e: MouseEvent) => {
+    const conditionToClose = !e.composedPath().includes(this.#dialogRef);
+    if (conditionToClose) {
```

Comment:
"`conditionToClose`" doesn't tell me **what** the condition is. In this case, the variable name must represent what the condition is. I already know that the variable is a boolean, but I don't know what the meaning is.

```suggestion
    const clickWasMadeOutsideTheDialog = !e.composedPath().includes(this.#dialogRef);
    
    if (clickWasMadeOutsideTheDialog) {
```